server/server_race_test.go

```
@@ -0,0 +1,181 @@
+
+	runConcurrentOperation(&wg, testDuration, "add-middleware", func() {
+		middleware := func(next ToolHandlerFunc) ToolHandlerFunc {
+			return func(ctx context.Context, req mcp.CallToolRequest) (*mcp.CallToolResult, error) {
+				return next(ctx, req)
+			}
+		}
+		WithToolHandlerMiddleware(middleware)(srv)
+	})
+
+	runConcurrentOperation(&wg, testDuration, "list-tools", func() {
+		srv.handleListTools(ctx, "123", mcp.ListToolsRequest{})
```

golangci-lint (1.64.8)

Error return value of `srv.handleListTools` is not checked

(errcheck)